### DISCUSSION TITLE
unable to build/run lnd app: error "unknown field ChanID in struct literal"

### DISCUSSION THREAD

**Discussion by 99Percent** [lightningnetwork/lnd#9317] [Troubleshooting] [answered]
I was using an outdate version of LND (0.15.3) and need to update to 0.18.3

However when I try to compile my testing lnd app I get the following error:

```
# github.com/lightningnetwork/lnd/invoices
../../go/pkg/mod/github.com/lightningnetwork/lnd@v0.18.3-beta/invoices/sql_store.go:1140:4: unknown field ChanID in struct literal of type sqlc.UpdateAMPSubInvoiceHTLCPreimageParams
```

This is at top of my go.mod file:
```
go 1.23.3

require (
	github.com/lightningnetwork/lnd v0.18.3-beta
	google.golang.org/grpc v1.59.0
	google.golang.org/protobuf v1.35.2
	gopkg.in/macaroon.v2 v2.1.0
)
```

Also I couldn't find a clear updated installation guide for LND development.

Working on regular Debian 12 amd64

**Comment by ViktorT-11** [comment]
Hi @99Percent! Check if there's an indirect dependency of `github.com/lightningnetwork/lnd/sqldb` in the `go.mod` file, and if updating that helps with the issue. `github.com/lightningnetwork/lnd/sqldb` `v1.0.5` is the latest tagged version, and I think either that or `v1.0.4` should hopefully work.

**Reply by 99Percent** [reply, marked as answer]
yes, I had 
```
	github.com/lightningnetwork/lnd/sqldb v1.0.3 // indirect
```

changed it to:
```
	github.com/lightningnetwork/lnd/sqldb v1.0.5 // indirect
```
And resolved the issue. Another error came up which was:
```
# github.com/lightningnetwork/lnd/lnrpc
../../go/pkg/mod/github.com/lightningnetwork/lnd@v0.18.3-beta/lnrpc/rpc_utils.go:32:3: unknown field UseHexForBytes in struct literal of type protojson.MarshalOptions
../../go/pkg/mod/github.com/lightningnetwork/lnd@v0.18.3-beta/lnrpc/rpc_utils.go:40:3: unknown field UseHexForBytes in struct literal of type protojson.UnmarshalOptions
```

but I fixed it by adding:
```
replace google.golang.org/protobuf => github.com/lightninglabs/protobuf-go-hex-display v1.30.0-hex-display
```
at the end of the go.mod file.

**Reply by ViktorT-11** [reply]
Awesome!

Just FYI, the latest tag for github.com/lightninglabs/protobuf-go-hex-display is [v1.34.2-hex-display](https://github.com/lightninglabs/protobuf-go-hex-display/releases/tag/v1.34.2-hex-display). But if `v1.30.0-hex-display` works for your app feel free to use that as well :)
